RELEASE.md

```diff
@@ -0,0 +1,221 @@
+a [Docker](https://www.docker.com) container for a hermetic release process.
+
+### Pre-requisites
+
+-   `docker`
+-   An account at [oss.sonatype.org](https://oss.sonatype.org/), that has
+    permissions to update artifacts in the `org.tensorflow` group. If your
+    account does not have permissions, then you'll need to ask someone who does
+    to [file a ticket](https://issues.sonatype.org/) to add to the permissions
+    ([sample ticket](https://issues.sonatype.org/browse/MVNCENTRAL-1637)).
+-   Optionally, an account at [bintray.com](https://bintray.com) that has permissions to
+    update the [tensorflow repository](https://bintray.com/google/tensorflow).
+    If your account does not have permissions, then you'll need to ask one of
+    the [organization administrators](https://bintray.com/google) to give you
+    permissions to update the `tensorflow` repository. Please keep the
+    [repository option](https://bintray.com/google/tensorflow/edit?tab=general)
+    to *"GPG sign uploaded files using Bintray's public/private key pair"*
+    **unchecked**, otherwise it will conflict with locally signed artifacts.
+-   A GPG signing key, required
+    [to sign the release artifacts](http://central.sonatype.org/pages/apache-maven.html#gpg-signed-components).
+
+### Preparing a release
+
+#### Major or minor release
+
+1.  Get a clean version of the source code by cloning the
+    [TensorFlow Java GitHub repository](https://github.com/tensorflow-java)
+    ```
+    git clone https://github.com/tensorflow/java
+    ```
+2.  Create a new branch for the release named `r<MajorVersion>.<MinorVersion>`
+    ```
+    git checkout -b r1.0
+    ```
+3.  Update the version of the Maven artifacts to the full version of the release
+    ```
+    mvn versions:set -DnewVersion=1.0.0
+    ``` 
+4.  Commit the changes and push the new branch to the GitHub repository
+    ```
+    git add .
+    git commit -m "Releasing 1.0.0"
+    git push --set-upstream upstream r1.0
+    ```
+
+#### Patch release
+
+1.  Get a clean version of the source code by cloning the
+    [TensorFlow Java GitHub repository](https://github.com/tensorflow-java)
+    ```
+    git clone https://github.com/tensorflow/java
+    ```
+2.  Switch to the release branch of the version to patch
+    ```
+    git checkout r1.0
+    ```
+3.  Patch the code with your changes. For example, changes could be merged from another branch you
+    were working on or be applied directly to this branch when the required changes are minimal.
+
+4.  Update the version of the Maven artifacts to the full version of the release
+    ```
+    mvn versions:set -DnewVersion=1.0.1
+    ``` 
+4.  Commit the changes and push the branch to the GitHub repository
+    ```
+    git add .
+    git commit -m "Releasing 1.0.1"
+    git push
+    ```
+
+### Building native artifacts
+
+Once a release branch (i.e. a branch prefixed by `r`) is pushed on the repository, a workflow will
```

Comment:
Does any commit to a release branch trigger this?

Reply:
Yes, it is assumed that any commit to a release branch is for releasing a version. On a major/minor release, a new branch is created. On a patch release, we push changes on the existing branch for the release we are patching.

Reply:
Ok. Might be worth explicitly noting that then.